bundles/org.eclipse.jface/src/org/eclipse/jface/resource/ImageDescriptor.java

```diff
@@ -205,25 +205,50 @@ public static ImageDescriptor createFromURLSupplier(boolean useMissingImage, Sup
 	}
 
 	/**
-	 * Convenient method to create an ImageDescriptor from an URI
+	 * Convenient method to create an ImageDescriptor from an URI.
 	 *
-	 * Delegates to ImageDescriptor createFromURL
+	 * Delegates to {@link ImageDescriptor#createFromURL(URL)}. <em>Important</em>
+	 * This method should only be used when it's guaranteed that the given
+	 * {@link URI} is also a valid {@link URL}, in order to avoid the
+	 * {@link MalformedURLException} thrown by {@link URI#toURL()}.
+	 *
+	 * If the URI is {@code null} or not a valid {@link URL}, then an image from
+	 * {@link #getMissingImageDescriptor()} will be returned.
 	 *
 	 * @param uriIconPath The URI of the image file.
 	 * @return a new image descriptor
 	 *
-	 * @since 3.19
+	 * @since 3.36
 	 */
-	public ImageDescriptor imageDescriptorFromURI(URI uriIconPath) {
+	public static ImageDescriptor createFromURI(URI uriIconPath) {
+		if (uriIconPath == null) {
+			return getMissingImageDescriptor();
+		}
 		try {
-			return ImageDescriptor.createFromURL(new URL(uriIconPath.toString()));
-		} catch (MalformedURLException | NullPointerException e) {
+			return ImageDescriptor.createFromURL(uriIconPath.toURL());
```

Comment:
We could also delegate the handling of the null case to `createFromURL()` and avoid coding the handling here again. But it's just a suggestion, I don't have a strong preference on it.
```suggestion
			return ImageDescriptor.createFromURL(uriIconPath != null ? uriIconPath.toURL() : null);
```

Reply:
I think it's better to move the null-handling to `createFromURL()` to really make sure that both methods behave the same on a `null` argument.

Reply:
Agree.